resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceTypeHelper.scala

```diff
@@ -0,0 +1,180 @@
+    }
+
+    if (!ResourceTypeHelper.isYarnResourceTypesAvailable()) {
+      logWarning(resourceTypesNotAvailableErrorMessage)
+      return resource
+    }
+
+    val resourceTypes = resourceTypesParam.map { case (k, v) => (
+      if (k.equals("memory")) {
+        logWarning("Trying to use 'memory' as a custom resource, converted it to 'memory-mb'")
+        "memory-mb"
+      } else k, v)
+    }
+
+    logDebug(s"Custom resource types: $resourceTypes")
+    resourceTypes.foreach { rt =>
+      val resourceName: String = rt._1
+      val (amount, unit) = getAmountAndUnit(rt._2)
+      logDebug(s"Registering resource with name: $resourceName, amount: $amount, unit: $unit")
+
+      try {
+        val resInfoClass = Utils.classForName(
+          "org.apache.hadoop.yarn.api.records.ResourceInformation")
+        val setResourceInformationMethod =
+          resource.getClass.getMethod("setResourceInformation", classOf[String],
+            resInfoClass)
+
+        val resourceInformation =
+          createResourceInformation(resourceName, amount, unit, resInfoClass)
+        setResourceInformationMethod.invoke(resource, resourceName, resourceInformation)
+      } catch {
+        case e: InvocationTargetException =>
+          if (e.getCause != null) {
+            throw e.getCause
+          } else {
+            throw e
+          }
+        case NonFatal(e) =>
+          logWarning(resourceTypesNotAvailableErrorMessage, e)
+      }
+    }
+    resource
+  }
+
+  def getCustomResourcesAsStrings(resource: Resource): String = {
+    if (resource == null) {
+      throw new IllegalArgumentException("Resource parameter should not be null!")
+    }
+
+    if (!ResourceTypeHelper.isYarnResourceTypesAvailable()) {
+      logWarning(resourceTypesNotAvailableErrorMessage)
+      return ""
+    }
+
+    var res: String = ""
+    try {
+      val resUtilsClass = Utils.classForName(
+        "org.apache.hadoop.yarn.util.resource.ResourceUtils")
+      val getNumberOfResourceTypesMethod = resUtilsClass.getMethod("getNumberOfKnownResourceTypes")
+      val numberOfResourceTypes: Int = getNumberOfResourceTypesMethod.invoke(null).asInstanceOf[Int]
+      val resourceClass = Utils.classForName(
+        "org.apache.hadoop.yarn.api.records.Resource")
+
+      // skip memory and vcores (index 0 and 1)
+      for (i <- 2 until numberOfResourceTypes) {
+        val getResourceInfoMethod = resourceClass.getMethod("getResourceInformation",
+          classOf[Int])
+        res ++= getResourceInfoMethod.invoke(resource, i.asInstanceOf[AnyRef]).toString()
+      }
+    } catch {
+      case e: InvocationTargetException =>
+        if (e.getCause != null) {
+          throw e.getCause
+        } else {
+          throw e
+        }
+      case NonFatal(e) =>
+        logWarning(resourceTypesNotAvailableErrorMessage, e)
+
+    }
+    res
+  }
+
+  def getAmountAndUnit(s: String): (Long, String) = {
```

Comment:
`private`, at least. But there's a single caller, why not just inline it?

Reply:
For the translation: 
I see, then the memory to memory-mb translation does not make sense, I get this now. 
Removed the translation. 
For getAmountAndUnit: I inlined the method